ci/deploy.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash -e
+#
+# Deploy your branch on VIP Go.
+#
+
+# This script uses various Circle CI and Travis CI environment
+# variables, Circle prefix their environment variables with
+# `CIRCLE_` and Travis with `TRAVIS_`.
+# Documentation:
+# https://circleci.com/docs/1.0/environment-variables/
+# https://docs.travis-ci.com/user/environment-variables/
+
+set -ex
+
+# The deploy suffix flexibility is mainly here to allow
+# us to test Circle and Travis builds simultaneously on
+# the https://github.com/Automattic/vip-go-skeleton/ repo.
+DEPLOY_SUFFIX="${VIP_DEPLOY_SUFFIX:--built}"
+
+BRANCH="${CIRCLE_BRANCH:-$TRAVIS_BRANCH}"
+
+SRC_DIR="${TRAVIS_BUILD_DIR:-$PWD}"
+BUILD_DIR="/tmp/vip-go-build-$(date +%s)"
+
+if [[ $CIRCLECI ]]; then
+	CIRCLE_REPO_SLUG="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}";
+fi
+REPO_SLUG=${CIRCLE_REPO_SLUG:-$TRAVIS_REPO_SLUG}
+REPO_SSH_URL="git@github.com:${REPO_SLUG}"
+COMMIT_SHA=${CIRCLE_SHA1:-$TRAVIS_COMMIT}
+DEPLOY_BRANCH="${BRANCH}${DEPLOY_SUFFIX}"
+cd $SRC_DIR
+COMMIT_AUTHOR_NAME="$( git log --format=%an -n 1 ${COMMIT_SHA} )"
+COMMIT_AUTHOR_EMAIL="$( git log --format=%ae -n 1 ${COMMIT_SHA} )"
+COMMIT_COMMITTER_NAME="$( git log --format=%cn -n 1 ${COMMIT_SHA} )"
+COMMIT_COMMITTER_EMAIL="$( git log --format=%ce -n 1 ${COMMIT_SHA} )"
+
+
+# Run some checks
+# ---------------
+
+if [[ -z "${BRANCH}" ]]; then
+	echo "No branch specified!"
```

Comment:
For this, and the other fatal errors below, it would be nice to include some instructions or direction on what to do next. Clearly something went wrong, but what do I do about it? (Related: https://www.smashingmagazine.com/2011/05/getting-started-with-defensive-web-design/)

Reply:
https://github.com/Automattic/vip-go-skeleton/pull/20/commits/4fbd17d7b66bf55344c2a45f96dd66a912b61c60